pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2773,3 +2774,26 @@ def test_merge_arrow_and_numpy_dtypes(dtype):
     result = df2.merge(df)
     expected = df2.copy()
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("tzinfo", [None, pytz.timezone("America/Chicago")])
+def test_merge_datetime_different_resolution(tzinfo):
+    # https://github.com/pandas-dev/pandas/issues/53200
+    df1 = DataFrame(
+        {
+            "t": [pd.Timestamp(2023, 5, 12, tzinfo=tzinfo)],
+            "a": [1],
+        }
+    )
+    df2 = df1.copy()
+    df2["t"] = df2["t"].dt.as_unit("s")
```

Comment:
Can we set the resolution for df1 explicitly as well? We might change the inference at some point which would make the test useless